qiime/core/type/primitive.py

```diff
@@ -349,6 +349,24 @@ def encode(self, value):
 Float = _Float('Float')
 
 
+class _Bool(_Primitive):
+    _valid_predicates = {Arguments,}
+
+    def _is_element_(self, value):
+        return isinstance(value, bool)
+
+    def decode(self, string):
+        if string not in ('false', 'true'):
+            raise TypeError("%s is neither 'true' or 'false'")
+
+        return string == 'true'
+
+    def encode(self, value):
+        return str(value)
```

Comment:
...so I'm okay with doing the `encode` as described but I guess I'm still confused as to why the enforcement for `Bool` but non-enforcement for the other primitives? Is the goal that `encode` is valid JSON without assurance that it is valid w.r.t. the type? All of these are valid but incorrect:

```python
In [1]: from qiime.core.type.primitive import Set, Float, Int

In [2]: Set.encode('foo')
Out[2]: '["f", "o", "o"]'

In [3]: Float.encode('foo')
Out[3]: 'foo'

In [4]: Int.encode('foo')
Out[4]: 'foo'
```